src/geff/metadata_schema.py

```diff
@@ -118,6 +121,47 @@ def axes_from_lists(
     return axes
 
 
+class RelatedObject(BaseModel):
+    type: Literal["labels", "image"] = Field(
+        ...,
+        description=(
+            "Type of the related object. 'labels' for label objects, 'image' for image objects."
+        ),
+    )
+    path: str = Field(
+        ...,
+        description=(
+            "Path of the related object within the zarr group, relative "
+            "to the zarr attributes file. "
+            "It is strongly recommended all related objects are stored as siblings "
+            "of the geff group within the top-level zarr group."
+        ),
+    )
+    label_prop: str | None = Field(
+        None,
+        description=(
+            "Property name for label objects. This is the property that will be used "
+            "to identify the labels in the related object. "
+            "This is only valid for type 'labels'."
+        ),
+    )
+
+    @model_validator(mode="after")
+    def _validate_model(self) -> Axis:
```

Comment:
The return type here should not be Axis but rather RelatedObject, right?

Reply:
Or sorry just copied and pasted, I'll fix it!